### ISSUE TITLE
amax and amin should return the full tensor with dim=()

### ISSUE THREAD

**Issue by asmeurer** [pytorch/pytorch#91922]
### 🐛 Describe the bug

```py
>>> import torch
>>> a = torch.asarray([[1., 2.], [3., 4.]])
>>> torch.amax(a, ())
tensor(4.)
```

Compare NumPy:

```
>>> import numpy as np
>>> a = np.asarray([[1., 2.], [3., 4.]])
>>> np.amax(a, ())
array([[1., 2.],
       [3., 4.]])
```

The second argument is supposed to be the dimensions to reduce, so when it is () it should reduce over no dimensions.

### Versions

Collecting environment information...
PyTorch version: 1.13.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.6.2 (x86_64)
GCC version: Could not collect
Clang version: 11.1.0
CMake version: version 3.25.1
Libc version: N/A

Python version: 3.9.15 | packaged by conda-forge | (main, Nov 22 2022, 08:55:37)  [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-12.6.2-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy==0.991
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.23.5
[pip3] numpydoc==1.5.0
[pip3] torch==1.13.0
[conda] cpuonly                   1.0                           0    pytorch-nightly
[conda] mkl                       2022.2.1         h44ed08c_16952    conda-forge
[conda] numpy                     1.23.5           py39hdfa1d0c_0    conda-forge
[conda] numpydoc                  1.5.0              pyhd8ed1ab_0    conda-forge
[conda] pytorch                   1.13.0          cpu_py39h0a103a1_0    conda-forge
[conda] torch                     1.11.0a0+git5347dab           dev_0    <develop>

cc @mruberry @rgommers

**Comment by asmeurer** [comment]
Also potentially relevant https://github.com/pytorch/pytorch/issues/58745 (although note that `amax` is closer to the array API spec than `max` because it allows tuples for `dim`).

**Comment by ngimel** [comment]
Duplicate of #29137, given bc-breaking implications this is unlikely to ever be changed.

**Comment by rgommers** [comment]
Duplicate indeed. The `dim=None` part of gh-29137 is the much more important part, and implementation of that is underway (done for the most heavily used reductions). 

@asmeurer I suspect that there are open issues for pretty much all the relevant differences between PyTorch and the array API standard. I probably still have those in memory, so please feel free to ping me directly for things like this.

Let's close this as a duplicate.
